### DISCUSSION TITLE
clustered stacked bar chart

### DISCUSSION THREAD

**Discussion by beliaev-maksim** [apache/superset#28181] [Ideas]
Hello,
Is there a plan to add clustered (grouped) stacked bar charts? Currently, we support either grouped or stacked charts, but not a combination of the two. Is it possible to implement a chart type that combines these features?

The idea is to create stacked bar charts while grouping them for visual simplification. This approach is especially helpful when you need to compare values both within a single bar and across a grouped cluster.

Here’s an example use case:
In this chart, we represent how each product (e.g., KTE, KRO, etc.) is performing. Stakeholders can easily focus on their respective products. Within each bar, we compare the quarter's target against the actual performance. Additionally, grouping bars by product allows us to visualize quarter-to-quarter trends in the context of the product.

![image](https://github.com/apache/superset/assets/51964909/e9890db0-cdd8-4e3e-81d9-0bb4674e741d)

**Comment by sfirke** [comment]
Agreed that this would be a great feature for Superset.  Here is a StackOverflow question asking for the same: https://stackoverflow.com/questions/75478746/can-i-have-a-bar-chart-that-is-both-stacked-and-grouped

**Reply by personofnorank** [reply]
Linked to https://github.com/apache/superset/discussions/30414 which I raised before I knew this had already been raised.

**Comment by oleg-savko** [comment]
+1 ultimately usefull feature, sadly for superset, that other bi tools have it...
I believe it will appear in the near future!)

**Comment by beliaev-maksim** [comment]
See follow up here: https://github.com/apache/superset/issues/32496
